lib/chef/audit/runner.rb

```diff
@@ -0,0 +1,244 @@
+
+require_relative "default_attributes"
+require_relative "fetcher/automate"
+require_relative "fetcher/chef_server"
+require_relative "reporter/audit_enforcer"
+require_relative "reporter/automate"
+require_relative "reporter/chef_server_automate"
+require_relative "reporter/json_file"
+
+class Chef
+  module Audit
+    class Runner < EventDispatch::Base
+      extend Forwardable
+
+      attr_reader :run_status
+      def_delegators :run_status, :node, :run_context
+      def_delegators :node, :logger
+
+      def enabled?
+        cookbook_collection = run_context.cookbook_collection
+
+        logger.info("#{self.class}##{__method__}: inspec profiles? #{inspec_profiles.any?}")
+        logger.info("#{self.class}##{__method__}: audit cookbook? #{cookbook_collection.key?("audit")}")
+
+        inspec_profiles.any? && !cookbook_collection.key?("audit")
+      end
+
+      def run_completed(_node, run_status)
+        @run_status = run_status
+
+        return unless enabled?
+
+        logger.info("#{self.class}##{__method__}: enabling audit mode")
+
+        report
+      end
+
+# TODO: handle error reports
+=begin
+      # Called at the end of a failed Chef run.
+      def run_failed(exception, run_status); end
+
+      # Called after Chef client has loaded the node data.
+      # Default and override attrs from roles have been computed, but not yet applied.
+      # Normal attrs from JSON have been added to the node.
+      def node_load_completed(node, expanded_run_list, config); end
+=end
+
+      ### Below code adapted from audit cookbook's files/default/handler/audit_report.rb
+
+      def report(report = generate_report)
+        if report.empty?
+          Chef::Log.error "Audit report was not generated properly, skipped reporting"
+          return
+        end
+
+        Array(audit_attributes["reporter"]).each do |reporter|
+          send_report(reporter, report)
+        end
+      end
+
+      def inspec_opts
+        waivers = Array(audit_attributes["waiver_file"]).select do |file|
+          return true if File.exist?(file)
+
+          Chef::Log.error "The specified InSpec waiver file #{file} is missing, skipping it..."
+          false
+        end
+
+        {
+          "report" => true,
+          "format" => "json-automate", # TODO: Remove this? Deprecation notice in audit cookbook
+          "reporter" => ["json-automate"],
+          "output" => audit_attributes["quiet"] ? ::File::NULL : STDOUT,
+          "logger" => Chef::Log,
+          backend_cache: audit_attributes["inspec_backend_cache"],
+          attributes: audit_attributes["attributes"],
+          waiver_file: waivers,
+          reporter_message_truncation: audit_attributes["result_message_limit"],
+          reporter_backtrace_inclusion: audit_attributes["result_include_backtrace"],
+        }
+      end
+
+      def audit_attributes
+        @audit_attributes ||= Chef::Audit::DefaultAttributes::DEFAULTS.merge(node["audit"] || {})
```

Comment:
you may want to be careful to use a Mash/VividMash for the attributes.

you also might consider before starting cookbook_loader to just push the defaults into the node.default VividMash.  since that is cleared, you may simply be able to construct a VividMash and save it there without even caring about merging?  although it might be nicer to still merge / deep_merge it into that VividMash.

Reply:
You should then just be able to access the attributes through `node['audit']`